yamllint/rules/float_values.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2022 contributors the the yamllint project
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Use this rule to limit the permitted values for floats.
+YAML permits three classes of float expressions:
+
+`[-+]?(\\.[0-9]+|[0-9]+(\\.[0-9]*)?)([eE][-+]?[0-9]+)?`
+`[-+]?(\\.inf|\\.Inf|\\.INF)`
+`\\.nan|\\.NaN|\\.NAN`
+
+.. rubric:: Options
+
+* Use ``require-numeral-before-decimal'`` to require floats to start
+    with a numeral (ex 0.0 instead of .0)
+* Use ``forbid-scientific-notation`` to forbid scientific notation
+* Use ``forbid-not-a-number`` to forbid NaN values
+* Use ``forbid-infinity`` to forbid infinite values
+
+.. rubric:: Default values (when enabled)
+
+.. code-block:: yaml
+
+ rules:
+   float-values:
+        require-numeral-before-decimal: false
+        forbid-scientific-notation: false
+        forbid-not-a-number: false
+        forbid-infinity: false
```

Comment:
I am not getting an error from `flake8`.  Is there maybe a configuration I am missing?

Reply:
Your example (and the others following below) use a 2-spaces indentation, but the third sublevel options (starting at `require-numeral-before-decimal`) have an indentation of 7 spaces (instead of 2 + 2 = 4):
```yaml
rules:
  float-values:
       require-numeral-before-decimal: false
       forbid-scientific-notation: false
       forbid-nan: false
       forbid-inf: false
```
Hence, this YAML content would report errors, if linted by yamllint.